### ISSUE TITLE
Create less segments for small files

### ISSUE THREAD

**Issue by muraty** [kaltura/nginx-vod-module#1486]
Hello.

Is it possible to have only one segment for a file for HLS? Or specify minimum size of a segment?

For context, we have subtitle files those are less than 1MB for a 2 hour video. If we segment them into streams with default duration which is `10s` (`vod_segment_duration`), clients do hundreds of requests in total to download all segments. Segmentation into streams works well for video files but small files cause hundreads of unnecessary requests.

If having only one segment or specifying minimum segment size are not options, what can we do to have less segments? I think increasing [vod_segment_duration](https://github.com/kaltura/nginx-vod-module#vod_segment_duration) might work but sounds a little bit tricky.

Thanks,
Murat.

**Comment by erankor** [comment]
Yeah, this is correct, it makes sense to use larger segments for subtitles. I thought about supporting it long time ago, but never got to it... 
If it really bothers you, you can probably work around it by using 2 locations - one location will have a vod_segment_duration of 10s (or whatever you want for video/audio), the other location will have vod_segment_duration of 60s (for example). You can use a regex on the subtitle location, so that it will capture only requests for subtitle index.m3u8/seg.vtt.

**Comment by muraty** [comment]
That's what I thought as a workaround. Thanks for the suggestion anyway.

A configuration for this might be a nice-to-have if possible. I don't know how much work it needs.
